### ISSUE TITLE
Alignment issues [Inputs, Popovers] (v1.4 03.10.2015)

### ISSUE THREAD

**Issue by juliyvchirkov** [timcreative/get-shit-done-pro#30]
greetings

I'd like to report about 2 alignment issues I've found in «Get shit done PRO» tutorial v1.4 03.10.2015 (please click on previews for the full screenshots)

***

**the first alignment issue is on x_get_shit_done_PRO/HTML/components.html page in the «Inputs» section x_get_shit_done_PRO/HTML/components.html#inputs-row**

[<img src="http://i.imgur.com/GJVWL4T.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/YaONZ3G.png)

as you can see on the screenshot above, the cross sign (typically named as &amp;times;) in the error field is aligned to the bottom of the field while it should be aligned to the middle

[<img src="http://i.imgur.com/JqUvTJr.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/wHj6hZl.png)

I beleive this inproper alignment is caused by incorrect font / classes used for the error field in this example. if you take the html code for the error field shown on the screenshot above

```
<div class="form-group has-error has-feedback">
    <input type="text" value="Error" class="form-control" />
    <span class="glyphicon glyphicon-remove form-control-feedback"></span>
</div>
```

[<img src="http://i.imgur.com/T8ioyfa.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/SNKAOSH.png)

and replace the third row ```<span class="glyphicon glyphicon-remove form-control-feedback"></span>``` w/ ```<span class="fa fa-times form-control-feedback"></span>```

i.e. if you change the font from «Glyphicons» to «FontAwesome», and apply corresponding classes ```fa fa-times``` designed for the «FontAwesome» instead of ```glyphicon glyphicon-remove``` designed for the «Glyphicons», so the whole html code for the error field will turn into 

```
<div class="form-group has-error has-feedback">
    <input type="text" value="Error" class="form-control" />
    <span class="fa fa-times form-control-feedback"></span>
</div>
```

you'll get the proper alignment of the cross sign in the error field, as shown on the screenshot above

[<img src="http://i.imgur.com/mqOsCXc.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/TgNnQKQ.png)

this sumption about font / classes is confirmed on another page of «Get shit done PRO» tutorial — x_get_shit_done_PRO/HTML/index.html in the «Inputs» section x_get_shit_done_PRO/HTML/index.html#inputs

as you can see on the screenshot above, there's no problem w/ proper alignment in the error field 'cause ```fa fa-times``` is used instead of ```glyphicon glyphicon-remove``` from scratch

***

**the second alignment issue is on x_get_shit_done_PRO/HTML/index.html page in the «Popovers» section x_get_shit_done_PRO/HTML/index.html#modals**

[<img src="http://i.imgur.com/40jPMDC.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/mca7cKC.png)

*popover on left*
very good
the arrow which binds this popover to the object is coloured w/ white according to the body of the popover where the arrow belongs

[<img src="http://i.imgur.com/7MoYd3w.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/XINsl3d.png)

*popover on top*
very good
the arrow which binds this popover to the object is coloured w/ white according to the body of the popover where the arrow belongs

[<img src="http://i.imgur.com/tkWKH0W.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/vM0nH0N.png)

*popover on bottom*
very good
the arrow which binds this popover to the object is coloured w/ azure according to the header of the popover where the arrow belongs

[<img src="http://i.imgur.com/5QZkXN8.png" title="Please click on this preview for the full screenshot">](http://i.imgur.com/9wj1PrW.png)

*popover on right*
not so good
the arrow which binds this popover to the object is coloured w/ white according to the body of the popover where the arrow should belong
but the half of this white arrow is aligned to the header of the popover which is coloured w/ azure
as it seems to me, this combo doesn't look nice

***

just in case — I get the reported above issues w/ «Get shit done PRO» v1.4 03.10.2015 on OS X 10.9.5 in Google Chrome 46.0.2490.86 (64-bit) & Safari 7.1.7 (9537.85.16.12) browsers

but I suppose these alignment issues does not depend on the specific operating system and browser

***

hope this quick report will help to make your great «Get shit done PRO» lib even better

regards,
Juliy V. Chirkov

**Comment by AzrielOmega** [comment]
The issue of the popover seem tricky to fix, because of the css generation of bootstrap 3.3.5 javascript. If the content length of the popover is 4 lines or over it'll be fine. But if it lower the issue seem to be unavoidable.

**Comment by alexandru-paduraru** [comment]
@juliyvchirkov thank you for the information, we put this issue and the other for the v1.4.2 release which will be in about 1-2 weeks. How @AzrielOmega said, the issue of the popover cannot be done without altering the javascript of bootstrap, we tried all the posibilities using the CSS tricks but nothing works... We will try again to find a solution for it.

**Comment by alexandru-paduraru** [comment]
Input alignment fixed in v1.4.2
